### DISCUSSION TITLE
Rules: Trigger for MQTT Traffic

### DISCUSSION THREAD

**Discussion by christian-ffm** [arendst/Tasmota#22375] [Support request] [answered]
Hello,
I would like to briefly switch on an LED each time an MQTT message has been successfully sent or retrieved. Unfortunately, I have no idea which event would be suitable as a trigger for a rule. Does anyone have an idea?
Thanks in advanced, best Chris

**Comment by sfromis** [comment, marked as answer]
Short answer: None such.

You can to some extent use wildcard triggers with `?` but lots of JSON processed for rules does not result in MQTT messages. You can to some extent filter with the prefix `tele-`, but many published MQTT messages are not telemetry. And a rule picking up something would almost by necessity trigger new MQTT messages, unless you suppress that by prefixing it with `?`. To some extent, you can have additional triggers filtering for "known cases", like power state switching, etc.

And no real way of triggering on commands arriving the MQTT subscription, but the command responses would result in MQTT messages.
